packages/react-admin/src/mui/input/ReferenceArrayInput.js

```diff
@@ -276,26 +286,31 @@ ReferenceArrayInput.defaultProps = {
     perPage: 25,
     sort: { field: 'id', order: 'DESC' },
     referenceRecords: [],
+    record: {},
```

Comment:
why this change? I'm afraid it may have side effects.

Reply:
The component expects a record in `componentWillReceiveProps` and it's not required, so a default value made sense to me, not sure about what side effects you're thinking about, but I can remove it in favor of extra checks in `componentWillReceiveProps` or making the prop required.